cmd/jujud/agent/machine/manifolds.go

```diff
@@ -695,10 +696,12 @@ func commonManifolds(config ManifoldsConfig) dependency.Manifolds {
 			Clock:                config.Clock,
 			Hub:                  config.CentralHub,
 			Logger:               loggo.GetLogger("juju.worker.dbaccessor"),
+			LogDir:               agentConfig.LogDir(),
```

Comment:
I think we should re-architect this a little.
- We only use the slow logger inside the node manager.
- We construct it outside the manager using the `loggo` logger, and `agentconfig.LogDir()`.
- Both of these are already passed to the node manager.

I would consider 
1) Threading the delegate all the way into the `NodeManager` constructor and materialising it inside or;
2) Creating a new composite logger encapsulating both the `loggo` and slow-log concerns passed in as a single constructor arg.

Reply:
3. We move the slow logger query as a dependency of the db accessor.

It feels like we're fighting to keep it outside of it. We might be better off embracing it.

Reply:
In the end, I chose 3. The code is more fitting with Juju and we don't introduce new concepts to the node manager.